src/main/java/org/cryptomator/cryptofs/common/MasterkeyBackupHelper.java

```diff
@@ -47,37 +41,40 @@ public static String generateFileIdSuffix(byte[] fileBytes) {
 
 	/**
 	 * Do a best-effort attempt to backup the masterkey at the given path. Fail silently if a valid backup already exists.
-	 * 
+	 *
 	 * @param masterKeyPath The masterkey file to backup
 	 * @throws IOException Any non-recoverable I/O exception that occurs during this attempt
 	 */
-	public static Path backupMasterKey(Path masterKeyPath) throws IOException {
+	public static Path attemptMasterKeyBackup(Path masterKeyPath) throws IOException {
 		byte[] keyFileContents = Files.readAllBytes(masterKeyPath);
 		String backupFileName = masterKeyPath.getFileName().toString() + generateFileIdSuffix(keyFileContents) + Constants.MASTERKEY_BACKUP_SUFFIX;
 		Path backupFilePath = masterKeyPath.resolveSibling(backupFileName);
-		try (WritableByteChannel ch = Files.newByteChannel(backupFilePath, StandardOpenOption.CREATE, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING)) {
+		try (WritableByteChannel ch = Files.newByteChannel(backupFilePath, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE)) {
 			ch.write(ByteBuffer.wrap(keyFileContents));
-		} catch (AccessDeniedException e) {
-			LOG.info("Storage device does not allow writing backup file. Comparing masterkey with backup directly.");
+		} catch (AccessDeniedException | FileAlreadyExistsException e) {
 			assertExistingBackupMatchesContent(backupFilePath, ByteBuffer.wrap(keyFileContents));
+		} catch (IOException e) {
+			LOG.warn("Failed to backup valid masterkey file.");
 		}
 		return backupFilePath;
 	}
-	
-	private static void assertExistingBackupMatchesContent(Path backupFilePath, ByteBuffer expectedContent) throws IOException {
+
+	private static void assertExistingBackupMatchesContent(Path backupFilePath, ByteBuffer expectedContent) {
 		if (Files.exists(backupFilePath)) {
 			// TODO replace by Files.mismatch() when using JDK > 12
```

Comment:
If you're already making changes in that method, maybe this TODO could be resolved? 😄